tools/analyzer_plugin/lib/src/async_plugin_apis/diagnostic.dart

```diff
@@ -195,7 +224,28 @@ class _DiagnosticGenerator {
     final filteredErrors = filterIgnores(
         collector.errors, unitResult.lineInfo, () => IgnoreInfo.forDart(unitResult.unit!, unitResult.content!));
 
-    return _GeneratorResult(filteredErrors, notifications);
+    final configuredErrors = configureErrorSeverity(filteredErrors);
+
+    return _GeneratorResult(configuredErrors, notifications);
+  }
+
+  List<AnalysisError> configureErrorSeverity(List<AnalysisError> errors) {
+    final configuredErrors = <AnalysisError>[];
+    for (final error in errors) {
+      if (_errorSeverityProvider.isCodeConfigured(error.code)) {
+        if (!_errorSeverityProvider.isCodeDisabled(error.code)) {
+          final severity = _errorSeverityProvider.severityForCode(error.code);
+          if (severity != null) {
+            error.severity = severity;
+          }
+          configuredErrors.add(error);
```

Comment:
#nit since we're mutating the errors in place, do we need to build a new list of them?

Reply:
Since we are removing some items from the list, I don't know a good way to do it in place. Removing items from a list while iterating over it is kind of wierd.